src/simtools/applications/derive_pulse_shape_parameters.py

```diff
@@ -0,0 +1,180 @@
+        --telescope LSTN-01 \
+        --model_version 7.0 \
+        --rise_width_ns 2.5 \
+        --fall_width_ns 5.0 \
+        --rise_range 0.1 0.9 \
+        --fall_range 0.9 0.1 \
+        --dt_ns 0.1 \
+        --time_margin_ns 10
+"""
+
+import logging
+
+import simtools.data_model.model_data_writer as writer
+from simtools.application_control import get_application_label, startup_application
+from simtools.configuration import configurator
+from simtools.model.model_utils import initialize_simulation_models
+from simtools.simtel.pulse_shapes import solve_sigma_tau_from_rise_fall
+from simtools.utils import names
+
+
+def _parse():
+    """Parse command line configuration for parameter derivation."""
+    config = configurator.Configurator(
+        label=get_application_label(__file__),
+        description=(
+            "Derive Gaussian sigma and exponential tau from rise/fall width specifications."
+        ),
+    )
+
+    config.parser.add_argument(
+        "--rise_width_ns",
+        help="Wdth on the rising edge in ns between rise_range fractions.",
+        type=float,
+        required=True,
+    )
+    config.parser.add_argument(
+        "--fall_width_ns",
+        help="Width on the falling edge in ns between fall_range fractions.",
+        type=float,
+        required=True,
+    )
+    config.parser.add_argument(
+        "--rise_range",
+        help="Fractional amplitudes (low high) for rise width, e.g. 0.1 0.9",
+        type=float,
+        nargs=2,
+        default=[0.1, 0.9],
+        required=False,
+    )
+    config.parser.add_argument(
+        "--fall_range",
+        help="Fractional amplitudes (high low) for fall width, e.g. 0.9 0.1",
+        type=float,
+        nargs=2,
+        default=[0.9, 0.1],
+        required=False,
+    )
+    config.parser.add_argument(
+        "--dt_ns",
+        help="Time sampling step in ns used by the solver.",
+        type=float,
+        default=0.1,
+        required=False,
+    )
+    config.parser.add_argument(
+        "--time_margin_ns",
+        help=(
+            "Margin (ns) added to both ends of the instrument readout window when deriving the "
+            "internal time window."
+        ),
+        type=float,
+        default=10.0,
+        required=False,
+    )
+
+    return config.initialize(
+        db_config=True, simulation_model=["telescope", "model_version"], output=True
+    )
+
+
+def main():
+    """Run parameter derivation and write results."""
+    app_context = startup_application(_parse)
+    log = logging.getLogger(__name__)
+
+    rise_width_ns = app_context.args["rise_width_ns"]
+    fall_width_ns = app_context.args["fall_width_ns"]
+    rise_range = tuple(app_context.args["rise_range"])
+    fall_range = tuple(app_context.args["fall_range"])
+    dt_ns = app_context.args["dt_ns"]
+    time_margin_ns = app_context.args["time_margin_ns"]
+    site = names.get_site_from_array_element_name(app_context.args["telescope"])
```

Comment:
This will not work for 'MSTx-FlashCam' or 'MSTx-NectarCam', which are valid for both North and South.

Read site from the command line: `site = app_context.args["site"]`.